superset/assets/src/dashboard/util/publishSubscriberUtil.js

```diff
@@ -0,0 +1,48 @@
+import * as _ from 'lodash';
+
+export const APPLY_FILTER = 'APPLY_FILTER';
+export const USE_AS_MODAL = 'USE_AS_MODAL';
+const CHART_UPDATE_SUCCEEDED = "CHART_UPDATE_SUCCEEDED"
+//check slice is modal or not
+export function isModalSlice(slice) {
+    let actions = getUniqueActionsForSlice(slice);
+    return isUseAsModalActionExist(actions);
+}
+
+export function isUseAsModalActionExist(actions) {
+    return (actions.indexOf(USE_AS_MODAL) != -1)
+}
+
+export function getUniqueActionsForSlice(slice) {
+    let actions = [];
+    let subscriberLayers = slice.form_data.subscriber_layers
+    if (subscriberLayers) {
+        subscriberLayers.forEach(element => {
+            actions = _.union(actions, element.actions);
+        });
+    }
+    return actions;
+}
+
+export function getModalSliceIDFor(publishSubscriberMap, publisherId) {
+    const subscribers = Object.values(publishSubscriberMap.subscribers);
+    var item = _.find(subscribers, function (item) {
+        return ((item.actions.indexOf(USE_AS_MODAL) > -1) && (Object.keys(item.linked_slices).indexOf(publisherId) != -1));
+    });
+    return item ? item.id : undefined
+}
+
+export function getUpdateSucceededChartId(items) {
```

Comment:
every where we are referring chart id to slice id. We should keep consistency.

Reply:
We should use slice sparingly as superset is start calling them Charts.

This has been down for all user facing documents but not inside codebase due to legacy issues. We should see what superset OSS community recommendes for newer changes

Reply:
Superset is using two type of Object for visualizations

1. chartPropShape : instance of this called as **chart** and it has more information than slice(slicePropShape) like chartStatus, Query,QueryResponse etc and it's mainly in memory object . 
2.slicePropShape : instance of this is called as **slice** and it is raw metadata of any visualization information and saved as metadata in DB. we use  it's information to create chart instance.

so name of any function and variable depends on  type of collection<Object> we are fetching info..and 
In ChartModal , mostly we are using chart instance every where.

I hope it answer.